public/app/features/search/page/SearchPage.tsx

```diff
@@ -0,0 +1,137 @@
+import { DataFrameView, LoadingState, NavModelItem } from '@grafana/data';
+import { Input } from '@grafana/ui';
+import React, { useMemo, useState } from 'react';
+import Page from 'app/core/components/Page/Page';
+import { useAsync } from 'react-use';
+import { buildStatsTable, filterDataFrame, getDashboardData } from './data';
+import { config, PanelRenderer } from '@grafana/runtime';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import { DashboardResult } from './types';
+
+export default function SearchPage() {
+  const node: NavModelItem = {
+    id: 'search',
+    text: 'Search',
+    icon: 'dashboard',
+    url: 'search',
+  };
+
+  const data = useAsync(getDashboardData, []);
+  const [query, setQuery] = useState('');
+  const filtered = useMemo(() => {
+    if (!data.value?.dashboards.length || !query.length) {
+      return data.value!; // everything
+    }
+    const dashboards = filterDataFrame(query, data.value.dashboards, 'Name', 'Description', 'Tags');
+    const panels = filterDataFrame(query, data.value.panels, 'Name', 'Description', 'Type');
+    return {
+      dashboards,
+      panels,
+      panelTypes: buildStatsTable(panels.fields.find((f) => f.name === 'Type')),
+      schemaVersions: buildStatsTable(dashboards.fields.find((f) => f.name === 'SchemaVersion')),
+    };
+  }, [query, data]);
+
+  if (!config.featureToggles.panelTitleSearch) {
+    return <div>Unsupported</div>;
+  }
+
+  return (
+    <Page navModel={{ node: node, main: node }}>
+      <Page.Contents>
+        <Input value={query} onChange={(e) => setQuery(e.currentTarget.value)} autoFocus spellCheck={false} />
+        <br /> <br />
+        {data.loading && <div>Loading....</div>}
+        {!data.loading && (
+          <div>
+            <AutoSizer style={{ width: '100%', height: '1500px' }}>
+              {({ width, height }) => {
+                // Helper... could be MUCH more efficient, but this approach lets us treat
+                // DataFrame results as well typed objects (the names must match the types)
+                const dashboards = new DataFrameView<DashboardResult>(filtered.dashboards);
+                return (
+                  <div>
```

Comment:
Probably can break this out into a separate component to clean this up a bit

Probably this can apply to other elements lower down as well

Reply:
Worked with @adela-almasan and broke this out into separate components into [latest commit](https://github.com/grafana/grafana/pull/45487/commits/7097dffc31c496451f4817edd1b637bdaa6cd6e7)